src/libraries/System.Security.Cryptography.Algorithms/tests/HmacMD5Tests.cs

```diff
@@ -21,63 +21,80 @@ public class HmacMD5Tests : Rfc2202HmacTests
             ByteUtils.RepeatByte(0xaa, 80),
         };
 
-        public HmacMD5Tests()
-            : base(s_testKeys2202)
+        private static readonly byte[][] s_testMacs2202 =
         {
-        }
+            null,
+            ByteUtils.HexToByteArray("9294727a3638bb1c13f48ef8158bfc9d"),
+            ByteUtils.HexToByteArray("750c783e6ab0b503eaa86e310a5db738"),
+            ByteUtils.HexToByteArray("56be34521d144c88dbb8c733f0e8b3f6"),
+            ByteUtils.HexToByteArray("697eaf0aca3a3aea3a75164746ffaa79"),
+            ByteUtils.HexToByteArray("56461ef2342edc00f9bab995690efd4c"),
+            ByteUtils.HexToByteArray("6b1ab7fe4bd7bf8f0b62e6ce61b9d0cd"),
+            ByteUtils.HexToByteArray("6f630fad67cda0ee1fb1f562db3aa53e"),
+        };
 
-        protected override HMAC Create()
+        public HmacMD5Tests()
+            : base(s_testKeys2202, s_testMacs2202)
         {
-            return new HMACMD5();
         }
 
-        protected override HashAlgorithm CreateHashAlgorithm()
-        {
-            return MD5.Create();
-        }
+        protected override int BlockSize => 64;
+        protected override int MacSize => 16;
+
+        protected override HMAC Create() => new HMACMD5();
+        protected override HashAlgorithm CreateHashAlgorithm() => MD5.Create();
+        protected override byte[] HashDataOneShot(byte[] key, byte[] source) =>
+            HMACMD5.HashData(key, source);
+
+        protected override byte[] HashDataOneShot(ReadOnlySpan<byte> key, ReadOnlySpan<byte> source) =>
+            HMACMD5.HashData(key, source);
+
+        protected override int HashDataOneShot(ReadOnlySpan<byte> key, ReadOnlySpan<byte> source, Span<byte> destination) =>
+            HMACMD5.HashData(key, source, destination);
 
-        protected override int BlockSize { get { return 64; } }
+        protected override bool TryHashDataOneShot(ReadOnlySpan<byte> key, ReadOnlySpan<byte> source, Span<byte> destination, out int written) =>
+            HMACMD5.TryHashData(key, source, destination, out written);
 
         [Fact]
         public void HmacMD5_Rfc2202_1()
         {
-            VerifyHmac(1, "9294727a3638bb1c13f48ef8158bfc9d");
+            VerifyHmac(1, s_testMacs2202[1]);
```

Comment:
IMHO this would have been more readable as single parametrized test. Then again, it is separate cleanup anyway.

Reply:
Yes, the tests could do with some cleanup. But I was trying my best to avoid re-writing the tests from scratch. Maybe I can do this in a follow up PR, unless someone feels strongly that "now is the time".

Reply:
I think I'm happy with what I came up with for non-crypto hashing tests (https://github.com/dotnet/runtime/pull/53623).  This could converge to something like that in a subsequent change.

I wouldn't want to hold up the API part on the test restructuring, though.